### PR TITLE
[FIXED JENKINS-41012] Actually get when branch/env to work

### PR DESCRIPTION
**Pull request by abayer**
[JENKINS-41012](https://issues.jenkins-ci.org/browse/JENKINS-41012)

Stop relying on the step context, since that wasn't actually getting
provided, and instead pass in the needed arguments from the *Script
class. Also make sure we actually have tests for all the conditions
*passing*, not just failing. Oy, but I feel silly there.

Assuming this is good, I'll do an 0.8.2 release first thing Thursday.

cc @reviewbybees esp @rsandell

### CODE REVIEW

File: pipeline-model-definition/src/test/resources/whenBranchFalse.groovy [hunk 1]
```diff
@@ -24,6 +24,9 @@
 
 pipeline {
     agent any
+    environment {
+        BRANCH_NAME = "master"
```
Comment (rsandell) [inline pipeline-model-definition/src/test/resources/whenBranchFalse.groovy]: Really? Is the environment not populated with the "from scm" settings that we use in the tests?

File: pipeline-model-definition/src/test/resources/whenEnvTrue.groovy [hunk 2]
```diff
@@ -0,0 +1,50 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016, CloudBees, Inc.
```
Comment (rsandell) [inline pipeline-model-definition/src/test/resources/whenEnvTrue.groovy]: 2017

File: pipeline-model-definition/src/test/resources/whenBranchFalse.groovy (hunk 1 above)
Comment (abayer) [inline pipeline-model-definition/src/test/resources/whenBranchFalse.groovy]: It is not. Sigh.

File: pipeline-model-definition/src/test/resources/whenEnvTrue.groovy (hunk 2 above)
Comment (abayer) [inline pipeline-model-definition/src/test/resources/whenEnvTrue.groovy]: Oy, copy paste!

File: pipeline-model-definition/src/test/resources/whenEnvFalse.groovy [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /*
  * The MIT License
  *
- * Copyright (c) 2016, CloudBees, Inc.
+ * Copyright (c) 2017, CloudBees, Inc.
```
Comment (rsandell) [inline pipeline-model-definition/src/test/resources/whenEnvFalse.groovy]: Well... the file only changed name, so the year shouldn't really be updated. But that's just a really _anal nit_ :)

Comment (abayer) [inline pipeline-model-definition/src/test/resources/whenEnvFalse.groovy]: Well, fixed. =)

Comment (abayer) [inline pipeline-model-definition/src/test/resources/whenEnvFalse.groovy]: Actually, not this one and OH GOD STOP WITH THE COPYRIGHTS I WILL CRY. =)

Comment (rsandell) [inline pipeline-model-definition/src/test/resources/whenEnvFalse.groovy]: Sorry, I used to work for an enterprise that made it's living on copyrights and patents, so I've been thoroughly indoctrinated on the matter ;)

File: pipeline-model-definition/src/test/resources/whenBranchFalse.groovy (hunk 1 above)
Comment (rsandell) [inline pipeline-model-definition/src/test/resources/whenBranchFalse.groovy]: Then perhaps we should be looking for some other way of figuring out what branch it is. But that could be a future enhancement.

**Comment by kuhnroyal** [comment]
@abayer does this somehow allow to match multiple patterns? `feature/**|PR*` or something like this?

**Comment by rsandell** [comment]
@kuhnroyal that would be #99

**Comment by kuhnroyal** [comment]
@rsandell thanks

**Pull request merged**